src/functional/iterator/embedded.py

```diff
@@ -600,11 +618,28 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
+    return axis is None
+
+
 def get_ordered_indices(
-    axes: Iterable[Axis], pos: Mapping[str, FieldIndex]
+    axes: Iterable[Axis], pos: Mapping[Tag, FieldIndex | SparsePositionEntry]
 ) -> tuple[FieldIndex, ...]:
-    assert all(axis.value in [*pos.keys()] for axis in axes)
-    return tuple(pos[axis.value] for axis in axes)
+    res: list[FieldIndex] = []
+    pos = deepcopy(pos)
+    for axis in axes:
+        if _is_tuple_axis(axis):
+            res.append(slice(None))
+        else:
+            assert axis is not None
+            assert axis.value in pos
+            elem = pos[axis.value]
+            if isinstance(elem, list):
+                res.append(elem.pop(0))
+            else:
+                assert isinstance(elem, (int, slice))
+                res.append(elem)
+    return tuple(res)
```

Comment:
Why is it needed to return a tuple here? It looks like it would be ok returning the list and annotating the return type as `Sequence` since `Sequence` is immutable and thus the static type should guarantee that the returned list is never modified.

Reply:
This is used to in `[]` of ndarray, which should be a tuple as far as I know.

Reply:
Ok, it makes sense.